### PR TITLE
fix: SKFP-1025 add compression of token before encrypt it

### PR DESCRIPTION
**Pull request by celinepelletier**

### CODE REVIEW

File: src/main/java/io/kidsfirst/core/service/StringCompressService.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    public static String compress(String str) {
+        if (str == null || str.isEmpty()) {
+            return str;
+        }
+
+        try {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            GZIPOutputStream gzip = new GZIPOutputStream(out);
+            gzip.write(str.getBytes());
+            gzip.close();
```
Comment (evans-g-crsj) [inline src/main/java/io/kidsfirst/core/service/StringCompressService.java:22]: possible leak if error

File: src/main/java/io/kidsfirst/core/service/StringCompressService.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    public static String decompress(String str) {
+        if (str == null || str.isEmpty()) {
+            return str;
+        }
+
+        try {
+            GZIPInputStream gis = new GZIPInputStream(new ByteArrayInputStream(str.getBytes(StandardCharsets.ISO_8859_1)));
+            BufferedReader bf = new BufferedReader(new InputStreamReader(gis, "ISO_8859_1"));
```
Comment (evans-g-crsj) [inline src/main/java/io/kidsfirst/core/service/StringCompressService.java:37]: no need to close anything?

**Review by evans-g-crsj — APPROVED** [review]

File: src/main/java/io/kidsfirst/core/service/AwsKmsService.java [hunk 3]
```diff
@@ -43,17 +43,23 @@ public Mono<String> encrypt(String original) {
                 return ByteBufferToString(bufferedCipher);
 
             } catch (UnsupportedEncodingException e) {
-                // Shouldn't be reachable, handle anyways
+                // Shouldn't be reachable, handle anyway
                 log.error(e.getMessage(), e);
                 return null;
             } catch (AWSKMSException e) {
-                log.error("AWSKMSException occurs when encrypting [{}] with message {}", original, e.getMessage());
+                log.error("AWSKMSException occurs when encrypting with message {}", e.getMessage());
```
Comment (evans-g-crsj) [inline src/main/java/io/kidsfirst/core/service/AwsKmsService.java:50]: are we sure we don t log secret in the message?

Comment (celinepelletier) [reply src/main/java/io/kidsfirst/core/service/AwsKmsService.java:50]: yes, the error message is 
```
1 validation error detected: Value at 'plaintext' failed to satisfy constraint: Member must have length less than or equal to 4096 (Service: AWSKMS; Status Code: 400; Error Code: ValidationException; Request ID: 6390d71d-ce86-4722-bf6f-705a84904edc; Proxy: null)
```

**Review by evans-g-crsj — APPROVED** [review]

**Pull request merged**